### PR TITLE
Support / in prefix for SNS topics

### PR DESCRIPTION
**Pull request by aheyman11**
Allow for `/` in prefix names when installing pipelines.

We add an attribute `sns_prefix` to the object store, which is the same as `s3.prefix` but with `/` replaced with `_`. Then we make sure to reference that attribute in all pipeline templates.

Do the same thing when creating SNS topics in `sns_subscribe.sh`.

### CODE REVIEW

File: python/etl/commands.py [hunk 1]
```diff
@@ -116,6 +116,7 @@ def run_arg_as_command(my_name="arthur.py"):
             setattr(args, "bucket_name", etl.config.get_config_value("object_store.s3.bucket_name"))
             if hasattr(args, "prefix"):
                 etl.config.set_config_value("object_store.s3.prefix", args.prefix)
+                etl.config.set_config_value("object_store.sns_prefix", args.prefix.replace('/', '_'))
```
Comment (tvogels01) [inline python/etl/commands.py]: Why should the SNS topic be part of the object store?

File: python/etl/templates/pizza_load_pipeline.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
         },
         {
             "id": "SNSParent",
-            "topicArn": "arn:aws:sns:${resources.VPC.region}:${resources.VPC.account}:redshift-etl-status_${object_store.s3.prefix}"
+            "topicArn": "arn:aws:sns:${resources.VPC.region}:${resources.VPC.account}:redshift-etl-status_${object_store.sns_prefix}"
```
Comment (tvogels01) [inline python/etl/templates/pizza_load_pipeline.json]: Shouldn't the topic for page get the same treatment?

Comment (aheyman11) [reply python/etl/templates/pizza_load_pipeline.json]: Yes indeed.

File: python/etl/commands.py (hunk 1 above)
Comment (aheyman11) [reply python/etl/commands.py]: No strong reason – would `resources.SNS.prefix` be better?

**Comment by bhtucker** [comment]
Seems reasonable under a name like `sns.suffix`.

This does diverge from how we've avoided this problem in `/current` and `/validation` cases, and how we sanitize table names for DynamoDB. I'd be more comfortable with this if we have a clear philosophy around nested prefixes in general.

To some extent this problem is emerging because of sharing one `object_store` across multiple Arthur projects, so is this problem a signal that we really shouldn't be doing that?

**Comment by tvogels01** [comment]
Indeed, the DynamoDB naming is what came to my mind.  We should sanitize the same way across AWS resources.

As for the object store, we could split this up for our two lines of work.  We should probably keep the same data lake and so make sure we don't step on each others toes there.  But then data lakes should be combined so that seems fair.

In general, I'm not thrilled about consistency of prefixes (or lack thereof). Maybe we do need to add the project in, say `dev` or `prod` or `tom` or `andrea`?  So the SNS topic would be `dw-etl-dev-status_development` where `dw-etl-dev` is shared across multiple resources to make it easy to search?  (So it would be the `dw-etl-dev-events_development` table in DynamoDB then.)

**Comment by tvogels01** [comment]
Discussions aside, I'd be happy to merge into v1.5 RC for now and we can go from there.

**Comment by aheyman11** [comment]
Up to you @tvogels01 on whether you want to merge this... I had to hack something in order to install pipelines using our shared object store, so I mainly put this PR up in case it was useful to others. Definitely open to waiting until we conform this with Dynamo.

**Comment by tvogels01** [comment]
I'm leaning towards making two changes.  One is to use `-` as the separator. Or maybe even re-use the code from DynamoDB which is a lot more paranoid.  The other change is to call this the "topic suffix" since SNS has more than topics so "sns suffix" is not descriptive enough.

**Pull request merged**